packages/react-components/react-menu-grid-preview/library/src/components/MenuGrid/useCheckMenuGridNesting.ts

```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+
+import { useCheckNesting } from '../../utils/useCheckNesting';
+
+export const useCheckMenuGridNesting = (ref: React.RefObject<HTMLElement>): void => {
```

Comment:
Instead of receiving a reference as a parameter for this hook, how about if this hook defines the reference? That way whatever calls this hook is only responsible for merging refs, and not for defining them

Reply:
i.e. the hook should return a ref instead of taking a ref in params

Reply:
Thanks, I have implemented the suggested change.